### PR TITLE
Issue #130 / Pool Reorder

### PR DESCRIPTION
**Pull request by blinkier**
Pools can be reordered/reprioritized through UI

### CODE REVIEW

File: miner.c [hunk 1]
```diff
@@ -4247,6 +4247,39 @@ static void display_pools(void)
...
+                        goto retry;
+                }
+                selected = curses_int("Select pool number");
+                if (selected < 0 || selected >= total_pools) {
+                        wlogprint("Invalid selection\n");
+                        goto retry;
+                }
+                priority = curses_int("Select new priority");
+                if (priority < 0 || priority >= total_pools) {
+                        wlogprint("Invalid selection\n");
```
Comment (luke-jr) [inline miner.c]: Perhaps "Invalid priority\n"?

**Comment by luke-jr** [comment]
This seems to actually try to move the pool in the pools array. That will cause all sorts of problems. Only pool->prio should be adjusted. Perhaps check out the start of switch_pools?

**Comment by blinkier** [comment]
Yes, it does reorder the pool array. The idea is to be able to then write the new order into a config file (since this is interpreted as priority on read in). The only thing that's perhaps counter intuitive is that if you give a pool priority 0, bfgminer doesn't immediately switch to it.

I've been using the code since 2.7.5 and haven't really had any issues with it. I looked, very briefly, at the pool switching strategies and didn't immediately see any problems. Did I miss something?

File: miner.c [hunk 2]
```diff
@@ -4247,6 +4247,39 @@ static void display_pools(void)
...
+                        for (i = selected; i < priority; i++)
+                                pools[i] = pools[i + 1];
+                } else {
+                        for (i = selected; i > priority; i--)
+                                pools[i] = pools[i - 1];
+                }
+                pools[priority] = pool;
+		for (i = 0; i < total_pools; i++)
+			pools[i]->pool_no = pools[i]->prio = i; 
+
```
Comment (blinkier) [inline miner.c]: Perhaps this makes sense?

```
if (selected == 0 || priority == 0)
    switch_pools(pools[0]);
```

**Comment by luke-jr** [comment]
Probably it would make sense to abstract and base this on the RPC poolpriority command (api.c:2189)

**Pull request merged**